src/tree/gpu_hist/row_partitioner.cu

```diff
@@ -159,25 +162,23 @@ void RowPartitioner::SortPositionAndCopy(const Segment& segment,
                                          bst_node_t right_nidx,
                                          int64_t* d_left_count,
                                          cudaStream_t stream) {
-  dh::TemporaryArray<bst_node_t> position_temp(position_a_.size());
-  dh::TemporaryArray<RowIndexT> ridx_temp(ridx_a_.size());
```

Comment:
I suspect building `TemporaryArray` at every invocation of `UpdatePosition` gets expensive in the particular example provided.